book/entities.rst

```diff
@@ -0,0 +1,1182 @@
+    documentations.
+
+.. _book-entities-config-annotation:
+
+The ``@Config`` Annotation
+..........................
+
+To make the ``Hotel`` entity from the first part of the chapter configurable, simply import the
+:class:`@Config <Oro\\Bundle\\EntityConfigBundle\\Metadata\\Annotation\\Config>` annotation and
+use it in the class docblock::
+
+    // src/Acme/DemoBundle/Entity/Hotel.php
+    namespace Acme\DemoBundle\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+    use Oro\Bundle\EntityConfigBundle\Metadata\Annotation\Config;
+
+    /**
+     * @ORM\Entity
+     * @ORM\Table(name="acme_hotel")
+     * @Config
+     */
+    class Hotel
+    {
+        // ...
+    }
+
+You can also change the default value of each configurable option using the ``defaultValues``
+argument::
+
+    // src/Acme/DemoBundle/Entity/Hotel.php
+    namespace Acme\DemoBundle\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+    use Oro\Bundle\EntityConfigBundle\Metadata\Annotation\Config;
+
+    /**
+     * @ORM\Entity
+     * @ORM\Table(name="acme_hotel")
+     * @Config(
+     *     defaultValues={
+     *         "acme_demo"={
+     *             "comment"="Our hotels"
+     *         }
+     *     }
+     * )
+     */
+    class Hotel
+    {
+        // ...
+    }
+
+.. _book-entities-config-field-annotation:
+
+The ``@ConfigField`` Annotation
+...............................
+
+Similarly the ``@Config`` annotation for entities, you can use the
+:class:`@ConfigField <Oro\\Bundle\\EntityConfigBundle\\Metadata\\Annotation\\ConfigField>`
+annotation to make properties of an entity configurable::
+
+    // src/Acme/DemoBundle/Entity/Hotel.php
+    namespace Acme\DemoBundle\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+    use Oro\Bundle\EntityConfigBundle\Metadata\Annotation\ConfigField;
+
+    /**
+     * @ORM\Entity
+     * @ORM\Table(name="acme_hotel")
+     */
+    class Hotel
+    {
+        // ...
+
+        /**
+        * @ORM\Column(type="string", length=255)
+        * @ConfigField
+        */
+        private $name;
+
+        // ...
+    }
+
+Default values can be changed in the same way as it can be done on the entity level::
+
+    // src/Acme/DemoBundle/Entity/Hotel.php
+    namespace Acme\DemoBundle\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+    use Oro\Bundle\EntityConfigBundle\Metadata\Annotation\ConfigField;
+
+    /**
+     * @ORM\Entity
+     * @ORM\Table(name="acme_hotel")
+     */
+    class Hotel
+    {
+        // ...
+
+        /**
+        * @ORM\Column(type="string", length=255)
+        * @ConfigField(
+        *     "defaultValues"={
+        *         "acme_demo"={
+        *             "auditable"=true
+        *         }
+        *     }
+        * )
+        */
+        private $name;
+
+        // ...
+    }
+
+Accessing the Entity Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that you know how you define additional configuration options and how to use them in your own
```

Comment:
actually config providers provide read-only access to the config options. It means that if you do `$container->get('oro_entity_config.provider.acme_demo')->get`